src/ApiCore/AuthWrapper.php

```diff
@@ -55,6 +61,54 @@ public function __construct(FetchAuthTokenInterface $fetchAuthTokenInterface, ca
         $this->authHttpHandler = $authHttpHandler;
     }
 
+    /**
+     * Factory method to create an AuthWrapper for a given set of scopes. If no credentials are
+     * provided in $keyFile or $keyFilePath, then ApplicationDefaultCredentials will be used.
+     *
+     * @param string[] $scopes The scopes required by this AuthWrapper.
+     * @param array $args {
+     *     @type string $keyFile
+     *           Optional. JSON credentials as an associative array.
+     *     @type string $keyFilePath
+     *           Optional. A JSON credential file path. If $keyFile is specified, $keyFilePath is ignored.
+     *     @type CacheItemPoolInterface $authCache
+     *           Optional. A cache for storing access tokens. Defaults to a simple in memory implementation.
+     * }
+     * @return AuthWrapper
+     * @throws ValidationException
+     */
+    public static function createWithScopes(array $scopes, array $args = [])
+    {
+        $args += [
+            'keyFile' => null,
+            'keyFilePath' => null,
+            'authCache' => null,
+        ];
+
+        $keyFile = $args['keyFile'] ?: $args['keyFilePath'];
+        $authCache = $args['authCache'] ?: new MemoryCacheItemPool();
+
+        try {
+            $authHttpHandler = HttpHandlerFactory::build();
+        } catch (\Exception $ex) {
+            throw new ValidationException("Failed to create authHttpHandler", $ex->getCode(), $ex);
```

Comment:
Regarding exception throwing - I don't have overly strong feelings about what it best. We define ApiException for exceptions caused by remote calls, and ValidationException for "local" exceptions. We could throw InvalidArgumentException in place of a lot of the ValidationException calls.

Regarding this case specifically - what is the consensus around throwing and catching the base Exception class? In general, it seems like bad practice, because any catch block will catch everything, even exceptions it can't handle. So I think there is an argument for converting base Exceptions into some more specific type.